cstar/base/component.py

```diff
@@ -158,3 +141,49 @@ def post_run(self) -> None:
         This abstract method will be implemented differently by different Component types.
         """
         pass
+
+
+class Discretization:
```

Comment:
Shouldn't this be an ABC too?
```suggestion
class Discretization(ABC):
```